vyper/ast/folding.py

```diff
@@ -255,21 +255,14 @@ def replace_constant(
     int
         Number of nodes that were replaced.
     """
-    is_struct = False
-
-    if isinstance(replacement_node, vy_ast.Call) and len(replacement_node.args) == 1:
-        if isinstance(replacement_node.args[0], vy_ast.Dict):
-            is_struct = True
-
     changed_nodes = 0
 
     for node in vyper_module.get_descendants(vy_ast.Name, {"id": id_}, reverse=True):
         parent = node.get_ancestor()
 
         if isinstance(parent, vy_ast.Call) and node == parent.func:
```

Comment:
My thinking is that this branch is only reachable in an expression like `ABC(...)`, given a constant struct like `ABC: StructName = StructName({..})`, but there should never be a situation where the former expression is a reference to the constant struct, and therefore `is_struct` will always be true in this branch.

Having said that, it seems that this branch itself could be redundant because it should not be possible for there to be a call like `Y(...)` and also a constant struct like `Y: StructName = StructName({...})` due to namespace collision?